### PR TITLE
api:  document MountPoint fields (swagger, godoc and docs)

### PR DESCRIPTION
**Pull request by thaJeztah**
follow-up to https://github.com/moby/moby/pull/42129, where I noticed that none of the fields had a description, nor an example value set.

Before:

Screenshot 2022-03-06 at 21 41 21

After:

Screenshot 2022-03-06 at 21 38 03

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -202,24 +202,72 @@ definitions:
...
+          Name is the name reference to the underlying data defined by `Source`
+          e.g., the volume name.
         type: "string"
+        example: "myvolume"
       Source:
+        description: |
+          Source location of the mount.
+
+          For volumes, this contains the storage location of the volume (within
+          `/var/lib/docker/volumes/`). For bind-mounts, and `npip`, this contains
```
Comment (tianon) [inline api/swagger.yaml:235]: `npip` :eyes:

(That's probably supposed to be `npipe`)

File: docs/api/v1.25.yaml [hunk 2]
```diff
@@ -170,24 +170,70 @@ definitions:
...
+          Name is the name reference to the underlying data defined by `Source`
+          e.g., the volume name.
         type: "string"
+        example: "myvolume"
       Source:
+        description: |
+          Source location of the mount.
+
+          For volumes, this contains the storage location of the volume (within
+          `/var/lib/docker/volumes/`). For bind-mounts, and `npip`, this contains
```
Comment (tianon) [inline docs/api/v1.25.yaml:201]: :eyes: (and all following)

File: api/types/types.go [hunk 3]
```diff
@@ -503,13 +503,44 @@ type DefaultNetworkSettings struct {
...
+	Type mount.Type `json:",omitempty"`
+
+	// Name is the name reference to the underlying data defined by `Source`
+	// e.g., the volume name.
+	Name string `json:",omitempty"`
+
+	// Source is the source location of the mount.
+	//
+	// For volumes, this contains the storage location of the volume (within
+	// `/var/lib/docker/volumes/`). For bind-mounts, and `npip`, this contains
```
Comment (tianon) [inline api/types/types.go:517]: :joy: :heart:

**Comment by thaJeztah** [comment]
Ah, dang! Autocorrect and tech-terminology still aren't friends (although I think I finally managed to train it to not "correct" JSON so Jason 🥳)

**Comment by thaJeztah** [comment]
`s/npip/npipe/` done 😂 😊  PTAL

**Review by tianon — APPROVED** [review]

**Review by ndeloof — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thanks for reviewing! Let me bring this one in

**Pull request merged**